### PR TITLE
:ghost: Tasking find refs

### PR DESCRIPTION
**Pull request by jortel**

### CODE REVIEW

File: task/manager.go [hunk 1]
```diff
@@ -389,6 +398,46 @@ func (m *Manager) disconnected(list []*Task) (kept []*Task, err error) {
...
+			err = &ExtensionNotFound{Name: name}
+			return
+		}
+	}
+	if task.Kind == "" {
+		return
+	}
+	kind, found := m.cluster.tasks[task.Kind]
+	if !found {
+		err = &ExtensionNotFound{Name: task.Kind}
```
Comment (mansam) [inline task/manager.go:427]: should be `KindNotFound`

File: importer/manager.go [hunk 2]
```diff
@@ -356,19 +356,12 @@ func (m *Manager) createApplication(imp *model.Import) (ok bool) {
 
 func (m *Manager) discover(application *model.Application) (err error) {
 	for _, kind := range Settings.Hub.Discovery.Tasks {
-		t := api.Task{}
+		t := &tasking.Task{}
```
[the remark this answers is not in the document]
Comment (jortel) [reply importer/manager.go]: Fixed.

File: task/manager.go (hunk 1 above)
Comment (jortel) [reply task/manager.go]: Fixed.

**Review by mansam — APPROVED** [review]
LGTM, seems to work well and the performance problem is resolved. Just minor nit about put/patch return codes.

**Comment by mansam** [comment]
Looks like the tests might need to be adjusted to deal with delete/cancel being fully aync.

**Comment by aufi** [comment]
E2E Test checkout is currently broken by my recent PR fixing diverged branches failures https://github.com/konveyor/go-konveyor-tests/pull/132

The error looks to be caused by just using main branch API tests in E2E test suite, instead of updated tests in this PR. I will update CI scripts, meanwhile in this case, feel free go ahead with merging.

**Review by aufi — APPROVED** [review]

**Pull request merged**